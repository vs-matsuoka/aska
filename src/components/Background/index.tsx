import Image from "next/future/image"

type BackgroundProps = {
  src: string,
}

function Background({ src }: BackgroundProps) {
  return (
    <div className="fixed">
      <div className="h-screen w-screen"></div>
      <Image
        src={src}
        alt="background"
        className="-z-50"
        fill
        style={{
          objectFit: 'cover'
        }}
      />
    </div>
  )
}

export default Background
