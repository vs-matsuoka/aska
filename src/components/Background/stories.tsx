import { Story, Meta } from '@storybook/react'
import Background from '.'

export default {
  title: 'Background',
  component: Background
} as Meta

export const Default: Story = () => <Background />
