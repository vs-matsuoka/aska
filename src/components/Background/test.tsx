import { render, screen } from '@testing-library/react';

import Background from '.';

describe('<Background />', () => {
  it('should render the heading', () => {
    const expectedSrc = '/blahblah';
    const { container } = render(<Background src={expectedSrc} />);

    expect(screen.getByAltText('background')).toBeInTheDocument();
    expect(screen.getByAltText('background')).toHaveAttribute(
      'src',
      `/_next/image?url=${encodeURIComponent(expectedSrc)}&w=3840&q=75`
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
