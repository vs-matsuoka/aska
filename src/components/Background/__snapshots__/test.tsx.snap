// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background /> should render the heading 1`] = `
<div
  class="fixed"
>
  <div
    class="h-screen w-screen"
  />
  <img
    alt="background"
    class="-z-50 select-none"
    data-nimg="fill"
    decoding="async"
    loading="lazy"
    sizes="100vw"
    src="/_next/image?url=%2Fblahblah&w=3840&q=100"
    srcset="/_next/image?url=%2Fblahblah&w=640&q=100 640w, /_next/image?url=%2Fblahblah&w=750&q=100 750w, /_next/image?url=%2Fblahblah&w=828&q=100 828w, /_next/image?url=%2Fblahblah&w=1080&q=100 1080w, /_next/image?url=%2Fblahblah&w=1200&q=100 1200w, /_next/image?url=%2Fblahblah&w=1920&q=100 1920w, /_next/image?url=%2Fblahblah&w=2048&q=100 2048w, /_next/image?url=%2Fblahblah&w=3840&q=100 3840w"
    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
  />
</div>
`;
